client_cli/src/main.rs

```diff
@@ -892,6 +906,130 @@ mod asset {
             Ok(())
         }
     }
+
+    #[derive(StructOpt, Debug)]
+    pub struct SetStore {
+        /// AssetId for the Store asset (in form of `asset##account@domain_name')
+        #[structopt(long = "asset-id")]
+        pub asset_id: AssetId,
+        /// The key for the store value
+        #[structopt(long)]
+        pub key: Name,
+        /// The value to be associated with the key
+        #[structopt(long)]
+        pub value: String,
```

Comment:
I suggest to use a bit different approach. `clap` can accept anything implementing `FromStr`.

I would suggest to create `ValueArg` (better name probably) which would implement parsing logic for `Value`.
This way we could have parsing shortcuts for some types and fallback to parsing json without json flag and without changing data model.

```rust 

struct ValueArg(Value);

impl FromStr for ValueArg {
    type Err = ...;

    fn from_str(s: &str) -> Result<Self, Self::Err> {
        // parse as bool
        
        // parse as ip

        // ..

        // if nothing worked parse as json
    }
}
```

And instead of String you can parse to `ValueArg` directly.